class/practice10/main.cpp

```diff
@@ -0,0 +1,201 @@
+#include <iostream>
+#include <vector>
+#include <string>
+#include <random>
+#include <print>
+#include <ranges>
+#include <algorithm>
+class RandomGen{
+    public:
+        int randomInt(int min, int max) {
+            std::random_device rd;
+            std::mt19937 gen(rd());
+            std::uniform_int_distribution<> dis(min, max);
+            return dis(gen);
+        }
+
+        double randomDouble(double min, double max) {
+            std::random_device rd;
+            std::mt19937 gen(rd());
+            std::uniform_real_distribution<> dis(min, max);
+            return dis(gen);
+        }
+};
+
+class Team{
+    public:
+        Team(std::string name) : name(name), goalsScored(0), goalsAgainst(0), yellowCards(0), redCards(0), topHeight(0), points(0) {}
+
+        int goalDelta() const {
+            return goalsScored - goalsAgainst;
+        }
+        //Setters
+        void setGoalsScored(int goals){
```

Comment:
classical situation with a big amount of setters and getters for every field. It can be used in that way but another option is to introduce some enum and template function. But I am not sure that it will improve the code quality so we can leave it as is